lib/pages/chat/chat.dart

```diff
@@ -1605,21 +1605,26 @@ class ChatController extends State<Chat>
         deltaBottomReversed > viewPortDimension - stickyTimestampHeight;
   }
 
+  Timer? _timestampTimer;
+
   void _handleHideStickyTimestamp() {
     Logs().d('Chat::_handleHideStickyTimestamp() - Hide sticky timestamp');
     stickyTimestampNotifier.value = null;
   }
 
-  void handleScrollEndNotification() async {
+  void handleScrollEndNotification() {
     Logs().d('Chat::handleScrollEndNotification() - End of scroll');
+    _timestampTimer?.cancel();
     if (PlatformInfos.isMobile) {
       _currentChatScrollState = ChatScrollState.endScroll;
-      await Future.delayed(_delayHideStickyTimestampHeader);
-      _handleHideStickyTimestamp();
+      _timestampTimer =
+          Timer(_delayHideStickyTimestampHeader, _handleHideStickyTimestamp);
```

Comment:
i have not used timer before, and usually go for .delayed, can you tell me more about the differences between two method ?

Reply:
As I understand, he will create a new `Timer` every time we got the `end` scrolling, and cancel it whenever starting scroll. Timer will start to run when it is init, then `handle***` when the time in timer reach.